taglib/mp4/mp4atom.cpp

```diff
@@ -97,19 +97,12 @@ MP4::Atom::Atom(File *file)
     if(name == c) {
       if(name == "meta") {
         offset_t posAfterMeta = file->tell();
-        ByteVector nextSize = file->readBlock(8).mid(4, 4);
         static constexpr std::array metaChildrenNames {
           "hdlr", "ilst", "mhdr", "ctry", "lang"
         };
-        bool metaIsFullAtom = true;
-        for(auto child : metaChildrenNames) {
-          if(nextSize == child) {
-            // meta is not a full atom (i.e. not followed by version, flags). It
-            // is followed by the size and type of the first child atom.
-            metaIsFullAtom = false;
-            break;
-          }
-        }
+        // meta is not a full atom (i.e. not followed by version, flags). It
+        // is followed by the size and type of the first child atom.
+        bool metaIsFullAtom = std::none_of(metaChildrenNames.begin(), metaChildrenNames.end(), [nextSize = file->readBlock(8).mid(4, 4)](const auto &child) { return nextSize == child; });
```

Comment:
Should we prefer `auto metaIsFullAtom` here? It is inconsistent with the suggestion using `find_if` in textidentificationframe.

Reply:
find_if and none_of have different return values. auto offers no advantage here.

Reply:
I understand that, the question I'm raising is whether stylistically we should prefer the use of `auto`. It seems to be the recommendation in https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines